### ISSUE TITLE
Display icons? 

### ISSUE THREAD

**Issue by markstos** [enkore/j4-dmenu-desktop#178]
Some dmenu-like utilities including Rofi and Fuzzel have the option display icons in dmenu mode. Both of those support the same protocol. Here's an example of using Fuzzel to display 3 entries in dmenu mode each with an icon:

```
echo -en  "Shutdown\0icon\x1fsystem-shutdown\nSleep\0icon\x1fsleep\nHibernate\0icon\x1fsystem-hibernate" | fuzzel --dmenu
```

I checked today and it appears that wofi, tofi and yofi don't support this.

On the other hand, Rofi and Fuzzel already natively support lauching desktop apps as well, but considering that "j4-dmenu-desktop" focuses on doing just one thing well, I could see this tool adding some advanced features focused on processing these desktop files that Rofi or Fuzzel don't support themselves.

**Comment by meator** [comment]
I don't think this feature would make sense for j4-dmenu-desktop. It is designed to be compatible with suckless's dmenu and with other programs which behave similarly to it. dmenu, j4-dmenu-desktop's default frontend (after which j4-dmenu-desktop got its name) doesn't support displaying icons.

I don't consider j4-dmenu-desktop to be a "complete" desktop file launcher. `Comment` field of desktop files is ignored (can't really do a tooltip in dmenu), no icons for desktop files are displayed, [application actions](https://specifications.freedesktop.org/desktop-entry-spec/1.5/extra-actions.html) are ignored, grouping of desktop files in accordance to [Desktop menu specification](https://specifications.freedesktop.org/menu-spec/latest/) is not implemented (see #131) etc. Some of these compromises were made because of technical limitations of dmenu, some of them were made to keep j4-dmenu-desktop fast and simple. j4-dmenu-desktop's goal never was to 100% support all relevant specifications, it focuses on what the user wants and it focuses on speed and correctness.

enkore, j4-dmenu-desktop's author, [has declared j4-dmenu-desktop feature complete](https://github.com/enkore/j4-dmenu-desktop/tree/r2.18?tab=readme-ov-file#project-status). Since I've taken over the project, I have remove this notice because I am open to changes and new features, but I am also satisfied with j4-dmenu-desktop's feature set.

You have hinted that the adoption of this protocol is low. Because of this and because of reasons outlined above, I don't think I'll be implementing this in the near future.

**Comment by markstos** [comment]
Thanks! I'll close this.
